File: java/34_Find_First_and_Last_Position_of_Element_in_Sorted_Array/34_Find_First_and_Last_Position_of_Element_in_Sorted_Array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] res = {-1,-1};
        if(nums == null || nums.length == 0) {
           	return res;
        }
        int left = 0;
        int right = nums.length - 1;
        while(left < right) {
            int mid = (left + (right - left) / 2);
            if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        if (nums[left] == target) {
            res[0] = left;
        }

        right = nums.length - 1;
        while(left < right) {
            int mid = (left + (right - left) / 2) + 1;
            if (nums[mid] > target) {
                right = mid - 1;
            } else {
                left = mid;
            }
        }
        if (nums[right] == target) {
            res[1] = right;
        }
        
        return res;
    }
}

File: go/34_Find_First_and_Last_Position_of_Element_in_Sorted_Array/34_Find_First_and_Last_Position_of_Element_in_Sorted_Array.go
func searchRange(nums []int, target int) []int {
    res := []int{-1, -1}
    if nums == nil || len(nums) == 0 {
        return res
    }
    left := 0
    right := len(nums) - 1
    for left < right {
        mid := (left + (right - left) / 2)
        if nums[mid] < target {
            left = mid + 1
        } else {
            right = mid
        }
    }
    if nums[left] == target {
        res[0] = left
    }
    right = len(nums) - 1
    for left < right {
        mid := (left + (right - left) / 2) + 1
        if nums[mid] > target {
            right = mid - 1
        } else {
            left = mid
        }
    }
    if nums[right] == target {
        res[1] = right
    }

    return res;
}
